test/ipns-pubsub.js

```diff
@@ -107,64 +108,43 @@ const subscribeToReceiveByPubsub = (nodeA, nodeB, idA, idB, callback) => {
   const topic = `${namespace}${base64url.encode(keys.routingKey.toBuffer())}`
 
   // try to resolve a unpublished record (will subscribe it)
-  nodeB.api.name.resolve(idA, (err) => {
+  try {
+    await nodeB.api.name.resolve(idA)
+  } catch (err) {
     expect(err).to.exist() // not found
 
-    series([
-      (cb) => waitForPeerToSubscribe(nodeB.api, topic, cb),
-      (cb) => nodeB.api.pubsub.subscribe(topic, checkMessage, cb),
-      (cb) => waitForNotificationOfSubscription(nodeA.api, topic, idB, cb),
-      (cb) => nodeA.api.name.publish(ipfsRef, { resolve: false }, cb),
-      (cb) => waitFor(() => subscribed === true, (50 * 1000), cb),
-      (cb) => nodeB.api.name.resolve(idA, cb)
-    ], (err, res) => {
-      expect(err).to.not.exist()
-      expect(res).to.exist()
-
-      expect(res[3].name).to.equal(idA) // Published to Node A ID
-      expect(res[5]).to.equal(ipfsRef)
-
-      callback()
-    })
-  })
+    await waitForPeerToSubscribe(nodeB.api, topic)
+    await nodeB.api.pubsub.subscribe(topic, checkMessage)
+    await waitForNotificationOfSubscription(nodeA.api, topic, idB)
+    const res1 = await nodeA.api.name.publish(ipfsRef, { resolve: false })
+    await waitFor(() => subscribed === true, (50 * 1000))
+    const res2 = await nodeB.api.name.resolve(idA)
+
+    expect(res1.name).to.equal(idA) // Published to Node A ID
+    expect(res2).to.equal(ipfsRef)
+  }
 }
 
 // wait until a peer know about other peer to subscribe a topic
-const waitForNotificationOfSubscription = (daemon, topic, peerId, callback) => {
-  retry({
-    times: 5,
-    interval: 2000
-  }, (next) => {
-    daemon.pubsub.peers(topic, (error, res) => {
-      if (error) {
-        return next(error)
-      }
-
-      if (!res || !res.length || !res.includes(peerId)) {
-        return next(new Error('Could not find peer subscribing'))
-      }
-
-      return next(null)
-    })
-  }, callback)
+const waitForNotificationOfSubscription = async (daemon, topic, peerId) => {
+  await pRetry(async () => {
```

Comment:
return?

Reply:
See earlier comment, we don't need to return anything from this function so let's not.